### PR TITLE
refactor(addRoute): create route only when parent.alias exists

### PR DESCRIPTION
**Pull request by alphawq**

### CODE REVIEW

**Review by posva — CHANGES_REQUESTED** [review]
Can you provide a reproduction or a failing test (before the change)?

**Comment by alphawq** [comment]
> Can you provide a reproduction or a failing test (before the change)?

It's not exactly a bug, but if parent.alias does not exist, there is no need to execute createRouteMap

File: src/create-matcher.js [hunk 1]
```diff
@@ -32,7 +32,7 @@ export function createMatcher (
     createRouteMap([route || parentOrRoute], pathList, pathMap, nameMap, parent)
 
     // add aliases of parent
-    if (parent) {
+    if (parent && parent.alias) {
```
Comment (posva) [inline src/create-matcher.js]: So you meant

```suggestion
    if (parent && parent.alias.length) {
```

? Otherwise it would always execute anyway

Comment (alphawq) [reply src/create-matcher.js:35]: oh~ yes,  you are right! Thanks for your suggestion.

Comment (alphawq) [reply src/create-matcher.js]: oh~ yes,  you are right! Thanks for your suggestion.

**Review by posva — APPROVED** [review]

**Pull request merged**